// aws-cdk/test/commands/test-resources/stacks/s3-stack.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoodTypeScriptStack = void 0;
const cdk = require("aws-cdk-lib");
const s3 = require("aws-cdk-lib/aws-s3");
/**
 * AWS CloudFormation Sample Template S3_Website_Bucket_With_Retain_On_Delete: Sample template showing how to create a publicly accessible S3 bucket configured for website access with a deletion policy of retain on delete.
 */
class GoodTypeScriptStack extends cdk.Stack {
    constructor(scope, id, props = {}) {
        super(scope, id, props);
        // Resources
        const s3Bucket = new s3.CfnBucket(this, 'S3Bucket', {
            accessControl: 'PublicRead',
            websiteConfiguration: {
                indexDocument: 'index.html',
                errorDocument: 'error.html',
            },
        });
        s3Bucket.cfnOptions.deletionPolicy = cdk.CfnDeletionPolicy.RETAIN;
        // Outputs
        this.websiteUrl = s3Bucket.attrWebsiteUrl;
        this.s3BucketSecureUrl = [
            'https://',
            s3Bucket.attrDomainName,
        ].join('');
    }
}
exports.GoodTypeScriptStack = GoodTypeScriptStack;
